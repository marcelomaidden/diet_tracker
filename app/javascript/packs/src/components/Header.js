import React from 'react';

const Header = ({ title }) => (
  <div className="background-blue d-flex p-4">
    <h1 className="h6 text-white mx-auto">{title}</h1>
  </div>
);

export default Header;