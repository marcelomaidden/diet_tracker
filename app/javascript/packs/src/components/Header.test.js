import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Header from './Header';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render a Header component', () => {
  act(() => {
    render(
      <Header title="Header test" />,
      container,
    );
  });
  expect(container.innerHTML).toMatchSnapshot();
});