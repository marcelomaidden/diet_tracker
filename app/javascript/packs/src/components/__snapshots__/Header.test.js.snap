// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Header component 1`] = `"<div class=\\"background-blue d-flex p-4\\"><h1 class=\\"h6 text-white mx-auto\\">Header test</h1></div>"`;
